### PR TITLE
require userdata loaded before running install

### LINKED ISSUE #696: :bug: kairos.service may start before cos-setup-network
When running in a cloud provider specifying a cloud config file (e.g. in openstack) the kairos.server (the installer) may start before the cos-setup-network service is done downloading cloud config files from the provider. In that case, our cloud-config is ignored and the installer continues by showing the QR code as if no cloud-config was provided.

Restarting the `kairos.service` with systemdctl makes it work (because now the cloud-config files are in place).

### PR DESCRIPTION
**Pull request by paynejacob**
Signed-off-by: Jacob Payne <jacob@spectrocloud.com>

Wait for userdata to load if the loading sentinel exists.

when PR gets merged)*:
Fixes #696

### CODE REVIEW

File: internal/agent/install.go [hunk 1]
```diff
@@ -102,11 +102,13 @@ func Install(dir ...string) error {
 		}
 	})
 
-	// Try to pull userdata once more. best-effort
-	if _, err := os.Stat("/oem/userdata"); err != nil {
-		if err := machine.ExecuteCloudConfig("/system/oem/00_datasource.yaml", "rootfs.before"); err != nil {
-			fmt.Println("Warning: Failed pulling from datasources")
+	for {
```
Comment (mudler) [inline internal/agent/install.go:105]: this is looking good! maybe we should timeout here and not let the loop go forever?

File: internal/agent/install.go [hunk 2]
```diff
@@ -102,11 +102,13 @@ func Install(dir ...string) error {
...
+		if _, err := os.Stat("/oem/.userdata_load"); os.IsNotExist(err) {
+			break
 		}
+
+		fmt.Println("userdata configuration has not yet completed.")
```
Comment (mudler) [inline internal/agent/install.go:110]: maybe we should write down what we are waiting for (for debugging reasons)

```suggestion
		fmt.Println("userdata configuration has not yet completed (waiting /oem/.userdata_load) .")
```

**Review by mudler — DISMISSED** [review]

**Comment by mudler** [comment]
giving it a re-thought. maybe we should move away from having the sentinel in `/oem`, but instead use `/run`?

File: internal/agent/install.go (hunk 1 above)
Comment (paynejacob) [reply internal/agent/install.go:105]: That is a good idea!

File: internal/agent/install.go (hunk 2 above)
Comment (paynejacob) [reply internal/agent/install.go:110]: I like the idea of more detail, ill give some context to what that file means as well.

**Review by Itxaka — APPROVED** [review]
niiice

**Pull request merged**